presto-proxy/src/test/java/com/facebook/presto/proxy/TestProxyServer.java

```diff
@@ -142,6 +142,26 @@ public void testQuery()
         }
     }
 
+    @Test
+    public void testSetSession()
```

Comment:
Can you explain how this test verifies the header changes?

Reply:
Headers are returned from the coordinator via the `X-Presto-Set-Session` header.  These headers are erroneously being serialized to `String` as `Collection<String>`, which caused them to be sent back from the coordinator as `[query_max_stage_count=10]` (singleton collection).  This test proves that the issue is fixed because we do not fail the query due to invalid session property name.